src/screens/claimables/ClaimingClaimableSharedUI.tsx

```diff
@@ -216,9 +218,13 @@ export const ClaimingClaimableSharedUI = ({
                   }
                 }}
                 onLongPress={() => {
-                  if (claimStatus === 'idle' || claimStatus === 'error') {
-                    setClaimStatus('claiming');
-                    claim();
+                  if (!isReadOnlyWallet || enableActionsOnReadOnlyWallet) {
+                    if (claimStatus === 'idle' || claimStatus === 'error') {
+                      setClaimStatus('claiming');
```

Comment:
I think we don't need this `setClaimStatus`, the `useMutation` has a `status`, we could hoist this logic of alerting on readOnly wallets to the parent and use this as just a presentation component